tests/unit/library/test_json.py

```diff
@@ -194,6 +194,82 @@ def test_bad_number(self, bad_string, good_bytes, failure_byte, allowed_bytes, m
             compact=compact,
         )
 
+class TestBoundedNumeric:
+    @pytest.mark.parametrize(
+        "instance, schema, should_pass",
+        [
+            # --- Integer type tests ---
+            (5, {"type": "integer", "minimum": 5}, True),
+            pytest.param(
+                *(5.0, {"type": "integer", "minimum": 5}, True),
+                marks=pytest.mark.xfail(reason="JSON technically allows trailing zeroes, but we currently don't")
+            ),
+            (5.1, {"type": "integer", "minimum": 5}, False),
```

Comment:
"Integer" and "5.1" .... is this telling me something about JSON schema that I don't want to know?

Reply:
Don't worry, this is a "should fail" case (see the False in the third slot of the tuple 😄)

Reply:
Probably should have made the comment on the previous line :-) The 'reason' gives me bad feelings about potentially subtle bugs (but we have to follow the JSON standard, so 🤷)